pkg/pb/referralmanager.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+option go_package = "pb";
+
+import "gogo.proto";
+import "google/protobuf/timestamp.proto";
+import "metainfo.proto";
+import "orders.proto";
+
+package referralmanager;
+
+// ReferralManager is a service for handling referrals
+service ReferralManager {
+  // GetTokens retrieves a list of unredeemed tokens for a user
+  rpc GetTokens(GetTokensRequest) returns (GetTokensResponse);
+  // ValidateToken validates a referral token from referral manager
+  rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
```

Comment:
Instead of `ValidateToken` and `SaveUserInfo`, what do you think about `ReserveToken` and `RedeemToken` or `BeginRedeemToken` and `FinishRedeemToken`/`CommitRedeemToken`?